### DISCUSSION TITLE
Q3 2022 post

### DISCUSSION THREAD

**Discussion by MabezDev** [MabezDev/blog#15] [Ideas]
Drop content to be mentioned in here :)

**Comment by georgik** [comment]
- https://wokwi.com/rust
- updates in https://github.com/esp-rs/awesome-esp-rust

**Reply by georgik** [reply]
espup - https://github.com/esp-rs/espup

**Comment by jessebraham** [comment]
This is trimmed down significantly but should cover the major changes. Feel free to prune it down further as needed. If there are any other project you'd like covered let me know, but most of them are fairly small and specific so I haven't bothered covering them.

### ESPFLASH

- Support for ESP32-C2 added [#204](https://github.com/esp-rs/espflash/pull/204)
- Support for using stub loaders added [#216](https://github.com/esp-rs/espflash/pull/216)
- Improved support for ESP-IDF partition tables (multiple PRs)
- Support for flashing the ESP32-S2 via CDC UART [#228](https://github.com/esp-rs/espflash/pull/228)
- Support for erasing OTA data [#229](https://github.com/esp-rs/espflash/pull/229)
- Support flashing via the built-in UART peripherals on Raspberry Pi [#234](https://github.com/esp-rs/espflash/pull/234)
- Relicensed under MIT/Apache-2.0 [#235](https://github.com/esp-rs/espflash/pull/235)
- CLI interface redesign [#239](https://github.com/esp-rs/espflash/pull/239)
- A large number of bug fixes, smaller features, refactoring, etc. (multiple PRs)
- Pushing towards a `2.0.0` release, most of the bigger tasks have been completed for this

### ESP-FLASHER-STUB

- Initial implementation has been completed for the ESP32-C3
    - https://github.com/esp-rs/esp-flasher-stub

### ESP-HAL

- Very basic support for running applications on the other core [#96](https://github.com/esp-rs/esp-hal/pull/96)
- ADC support [#97](https://github.com/esp-rs/esp-hal/pull/97) [#98](https://github.com/esp-rs/esp-hal/pull/98)
- Allow for configuring the UART [#99](https://github.com/esp-rs/esp-hal/pull/99) [#102](https://github.com/esp-rs/esp-hal/pull/102)
- Vectored interrupts [#103](https://github.com/esp-rs/esp-hal/pull/103) [#118](https://github.com/esp-rs/esp-hal/pull/118)
- Direct-boot support for the ESP32-S3 [#107](https://github.com/esp-rs/esp-hal/pull/107)
- CPU clock settings [#110](https://github.com/esp-rs/esp-hal/pull/110) [#111](https://github.com/esp-rs/esp-hal/pull/111)
- LEDC support [#114](https://github.com/esp-rs/esp-hal/pull/114)
- Critical section implementations [#151](https://github.com/esp-rs/esp-hal/pull/151)
- `embedded_hal@1.0.0-alpha.x` traits implemented for most peripherals (multiple PRs)
- A large number of bug fixes, smaller features, refactoring, etc. (multiple PRs)
- Had our first (and second) releases of each chip's HAL (newest versions are `esp32@0.5.0`, `esp32xx@0.2.0`)

**Comment by bjoernQ** [comment]
## ESP-WIFI

- WiFi support for ESP32-S3
- WiFi support for ESP32-S2

## ESP-STORAGE

- Available on crates.io
- Supports ESP32, ESP32-C3, ESP32-S2, ESP32-S3

## ESP-HAL

- Very first experimental steps to get DMA support (ESP32-C3, SPI for now)

_I would love to mention esp-serial-dbg but as long as I'm the only person to get it running, we shouldn't mention it 😄_
